plutus-tx-plugin/src/PlutusTx/Plugin.hs

```diff
@@ -120,15 +131,40 @@ mkSimplPass flags =
 parsePluginArgs :: [GHC.CommandLineOption] -> GHC.CoreM PluginOptions
 parsePluginArgs args = do
     let opts = PluginOptions {
-            poDoTypecheck = notElem "dont-typecheck" args
-            , poDeferErrors = elem "defer-errors" args
-            , poContextLevel = if elem "no-context" args then 0 else if elem "debug-context" args then 3 else 1
-            , poDumpPir = elem "dump-pir" args
-            , poDumpPlc = elem "dump-plc" args
-            , poOptimize = notElem "dont-optimize" args
+            poDoTypecheck = notElem' "dont-typecheck"
+            , poDeferErrors = elem' "defer-errors"
+            , poContextLevel = if elem' "no-context" then 0 else if elem "debug-context" args then 3 else 1
+            , poDumpPir = elem' "dump-pir"
+            , poDumpPlc = elem' "dump-plc"
+            , poOptimize = notElem' "dont-optimize"
```

Comment:
Should probably make the existing `dont-` options into `no-` options, but we can do that later.